python/pyarrow/tensorflow/plasma_op.cc

```diff
@@ -33,6 +33,7 @@
 #include "arrow/adapters/tensorflow/convert.h"
 #include "arrow/api.h"
 #include "arrow/io/api.h"
+#include "arrow/util/logging.h"
```

Comment:
@pcmoritz Would you know why `plasma_op.cc` includes Tensorflow's `logging.h` above? It produces a collision in the `DCHECK` macros.

Reply:
where is the DCHECK collision occurring? Are we clobbering their macro or vice versa? We should use ARROW_CHECK (or add ARROW_DCHECK_* aliases) so we can use these in an environment where there is ambiguity. We already disallow DCHECK in public headers